File: Current Courses/Plain Driven Programming/Lab/Assignment 3/human.py
class Human:
    """
    A class to represent a Human

    Attributes
    ----------
    first_name: str
        Player's first name
    last_name: str
        Player's last name
    age: int
        Player's age

    Methods
    -------
    set_first_name() -> None
        Updates human's first name
    set_last_name() -> None
        Updates human's last name
    set_age() -> None
        Updates human's age
    get_first_name -> str
        Returns human's first name
    get_last_name -> str
        Returns human's last name
    get_age -> int
        Returns human's age   
    introduce_self() -> None
        Prints human's info
    """
    
    def __init__(self, first_name: str, last_name: str, age: int) -> None:
        """
        Constructor for Human class

        Parameters
        ----------
        first_name: str
            Human's first name
        last_name: str
            Human's last name
        age: int
            Human's age
        """
        self.first_name = first_name
        self.last_name = last_name
        self.age = age

    def set_first_name(self, name: str) -> None:
        """
        Updates human's first name

        Args:
            name (str): human's first name
        """
        if self.first_name is None:
            self.first_name = name

    def set_last_name(self, name: str) -> None:
        """
        Updates human's last name

        Args:
            name (str): human's last name
        """
        if self.last_name is None:
            self.last_name = name

    def set_age(self, num: int) -> None:
        """
        Updates human's age

        Args:
            num (int): human's age
        """
        if self.age is None:
            self.age = num

    @property
    def get_first_name(self) -> str:
        """
        Returns human's first name

        Returns:
            str: human's first name
        """
        return self.first_name

    @property
    def get_last_name(self) -> str:
        """
        Returns human's last name

        Returns:
            str: human's last name
        """
        return self.last_name

    @property
    def get_age(self) -> int:
        """
        Returns human's age

        Returns:
            int: human's age
        """
        return self.age

    def introduce_self(self) -> None:
        """
        Prints human's info
        """
        print(f"Player's name:\t{self.first_name} {self.last_name}")
        print(f"Player's age:\t{self.age}")

File: Current Courses/Agile Programming/Assignments/Assignment Singleton/simple_hearts_game_singleton/bin/human.dart
/// An abstract class to represent a human, which has:
/// attributes for human's first and last name and age,
/// attribute getters,
/// methods.
abstract class Human {
  late String _firstName; // class attribute
  late String _lastName; // class attribute
  late int _age; // class attribute

  Human(); // Constructor

  String get firstName => this._firstName; // getter for _firstName
  set firstName(String value) =>
      this._firstName = value; // setter for _firstName

  String get lastName => this._lastName; // getter for _lastName
  set lastName(String value) => this._lastName = value; // setter for _lastName

  int get age => this._age; // getter for _age
  set age(int value) => this._age = value; // setter for _age

  /// Returns a formmated string containing classes attributes.
  @override
  String toString() {
    return 'Introducing: $_firstName $_lastName, $_age';
  }

  /// Introduces human.
  void introduceSelf() {
    print(toString());
  }
}
